lazy_tensor_core/lazy_tensor_core/csrc/ops/get_dimensions_size.cpp

```diff
@@ -17,10 +17,7 @@ GetDimensionsSize::GetDimensionsSize(
            lazy_tensors::ShapeUtil::MakeShape(
                GetShapeDimensionType(/*device=*/nullptr), {}),
            /*num_outputs=*/1, lazy_tensors::util::MHash(dimensions)),
-      dimensions_(std::move(dimensions)) {
-  SetShapeDeferred(
-      [&]() { return compiler::NodeLowering::Get()->Infer(this); });
-}
+      dimensions_(std::move(dimensions)) {}
```

Comment:
I wouldn't call it redundant, but it has the same purpose - it computes the output shape. The shape inference has to be deferred in the same manner as the lowering, since neither is provided by the core.

This is a stop-gap solution to this problem: https://github.com/pytorch/pytorch/blob/d18ba4b0305091b49be631558fc39127042b28f9/lazy_tensor_core/lazy_tensor_core/csrc/compiler/node_lowering.h#L21. It's a pretty high priority item to address, but I didn't have time for it yet.